apps/wakunode2/wakunode2_validator_signed.nim

```diff
@@ -29,9 +33,26 @@ proc msgHash*(pubSubTopic: string, msg: WakuMessage): array[32, byte] =
   ctx.update(pubsubTopic.toBytes())
   ctx.update(msg.payload)
   ctx.update(msg.contentTopic.toBytes())
+  ctx.update(msg.timestamp.uint64.toBytes(Endianness.littleEndian))
+  ctx.update(if msg.ephemeral: @[1.byte] else: @[0.byte])
 
   return ctx.finish()
 
+proc withinTimeWindow*(msg: WakuMessage): bool =
+  # Returns true if the message timestamp is:
+  # abs(now - msg.timestamp) < MessageWindowInSec
+  let ts = msg.timestamp
+  let now = getNanosecondTime(getTime().toUnixFloat())
```

Comment:
`waku_core` (exporting `time`) already provides `getNowInNanosecondTime()` so you don't have to import `std/times`

Reply:
yes right, i keep forgeting.

fixed 3092a519137b5db1663a479d6562da6ecfe98cac